### DISCUSSION TITLE
Add support for WSGI apps using write callables

### DISCUSSION THREAD

**Discussion by Cito** [encode/httpx#2921] [Ideas]
Old WSGI apps and frameworks sometimes make use of the legacy feature of sending output via a [write callable](https://peps.python.org/pep-3333/#the-write-callable) instead of returning it as the iterable output.

Currently, the WSGI support of HTTPX does not support these kind of applications. I suggest adding such support, because it is very simple to implement and some of the old frameworks are still used.

I have already provided a [PR](https://github.com/encode/httpx/pull/2920) to show how I imagine this support to look like.

**Comment by lovelydinosaur** [comment]
> I suggest adding such support, because it is very simple to implement and some of the old frameworks are still used.

Shall we start with this conversation or the pull request?

**Reply by Cito** [reply]
I have now also added a small real-world usage example in the PR. But we can also discuss here.

The PR is very cheap to implement and test, with no side effects. The only argument against is it that the feature is considered deprecated and should not be used any more by modern frameworks. Supporting legacy features, however small, is always an additional burden. I'm not sure how many apps and frameworks are still around that use this feature. But people may wonder why the WSGI support of httpx does not work *if* they work with one of these apps, maybe without even knowing.
